back/app/services/side_fx_initiative_service.rb

```diff
@@ -79,7 +79,11 @@ def add_autovote(initiative)
   end
 
   def log_activity_jobs_after_published(initiative, user)
-    LogActivityJob.set(wait: 20.seconds).perform_later(initiative, 'published', user, initiative.published_at.to_i)
+    LogActivityJob.set(wait: 20.seconds).perform_later(initiative, 'published', user_for_activity(initiative, user), initiative.published_at.to_i)
+  end
+
+  def user_for_activity(initiative, user)
```

Comment:
If we do need this method then maybe it could be generic for each different model and included in `SideFxHelper`?

Reply:
I was thinking about it, but it's very specific to posts that can be anonymous. So if we would go that way, we would either have to check for the type of the item or if the item supports the `anonymous?` predicate. Both solutions felt a bit like a red flag.